### PR TITLE
Feature: gist load & share

### PR DESCRIPTION
**Pull request by smacker**
Based on https://github.com/bblfsh/dashboard/pull/48 and https://github.com/bblfsh/dashboard/pull/71

Fixes issue https://github.com/bblfsh/dashboard/issues/46

### CODE REVIEW

**Comment by bzz** [comment]
@smacker quick question on status: if this is something WIP, could you add `WIP:` prefix to the title?

And one thing I find useful - if CI is failing, it worth extra step to communicate in PR comments why is that and if anybody is already looking into that.

**Comment by smacker** [comment]
@bzz it's finished. Everything works. I'm checking why CI has failed.

File: server/server.go [hunk 1]
```diff
@@ -88,6 +91,25 @@ func (s *Server) clientForRequest(req parseRequest) (protocol.ProtocolServiceCli
 	return newClient(req.ServerURL)
 }
 
+func (s *Server) LoadGist(ctx *gin.Context) {
```
Comment (smacker) [inline server/server.go]: I decided to keep it for now. Maybe it will be useful in near future. If not, we always can remove it.

**Comment by smacker** [comment]
I squashed commits because they had WIP in title and many things were changed after. So the history didn't make so much sense already.

**Pull request merged**